src/player.cpp

```diff
@@ -10884,42 +10884,45 @@ bool player::has_enough_charges( const item &it, bool show_msg ) const
     return true;
 }
 
-bool player::consume_charges(item *used, long charges_used)
+bool player::consume_charges( item& used, long qty )
 {
-    // Non-tools can use charges too - when they're comestibles
-    const auto tool = dynamic_cast<const it_tool*>(used->type);
-    const auto comest = dynamic_cast<const it_comest*>(used->type);
-    if( charges_used <= 0 || (tool == nullptr && comest == nullptr) ) {
+    if( !has_item( &used ) ) {
+        debugmsg( "Tried to consume charges for an item not in players possession" );
```

Comment:
So that check is there to protect the later usage of `i_rem`.

The question is should we be fixing this function (check for possession immediately prior to i_rem` or change those callers not to use a temporary object.

The former is more expedient, the latter is arguably more correct?